merlin-sdk/src/main/java/gov/nasa/jpl/aerie/merlin/protocol/types/Duration.java

```diff
@@ -494,11 +494,11 @@ public boolean equals(final Object o) {
   }
 
   public static Duration fromISO8601String(final String iso8601String){
-    return of(java.time.Duration.parse(iso8601String).toMillis(), MILLISECOND);
+    return of(java.time.Duration.parse(iso8601String).toNanos() / 1000, MICROSECONDS);
   }
 
   public String toISO8601String() {
-    return java.time.Duration.ofMillis(dividedBy(Duration.MILLISECONDS)).toString();
+    return java.time.Duration.ofNanos(this.durationInMicroseconds * 1000).toString();
```

Comment:
It occurs to me that multiplying by 1000 could actually cause arithmetic overflow in extreme cases -- a 64-bit integer can only span ~600 years (between approx. -300 to 300) so any time beyond that horizon is going to get slammed against the ceiling.

(Back when we defined `Duration`, we chose micros for our time resolution instead of nanos because a 300-year positive time horizon seemed somehow too short. I think this is because I hoped (and still hope) that the core modeling and simulation functionality could be used for much longer time-horizon applications.

This issue doesn't affect the replacement of `toISO8601String` in a later commit, since that outputs micros directly. And `fromISO8601String` (and its equivalent that's been moved to `PreparedStatements`) will "only" cause a loss of precision, so distinct times occurring within the same microsecond would be treated as simultaneous -- but this could be fixed by just storing a number of microseconds in the database. (Postgres' `interval` type is much more calendar-sensitive than we really want here, in any case.)

Reply:
This is a good point - I've updated the first commit to avoid representing the whole duration as a long in nanoseconds.

Now that you bring it up - I think `fromISO8601String` still has the overflow vulnerability if the duration is too large to represent as a long in nanoseconds - simply calling `toNanos` will throw. From the `java.time.Duration::toNanos` javadoc:
> If this duration is too large to fit in a long nanoseconds, then an exception is thrown.

I've propagated the update to the last commit - so that we convert the `seconds` and `nanos` components of a Duration to microseconds separately - if we overflow in microseconds, that's to be expected, but we won't overflow in nanoseconds (if I did my math right 🤞 )

Reply:
Yikes. Let's look at the trade-offs of replacing `interval` in Postgres soon >_<